catalog/hyperledger/single-cluster.bom

```diff
@@ -1,415 +1,292 @@
-        STATUS=$(sudo docker inspect -f {{.State.Running}} $(cat hyperledger.peer.id))
-        sudo service docker status && [ "${STATUS}" = "true" ]
-
-      brooklyn.initializers:
-      - type: org.apache.brooklyn.core.effector.ssh.SshCommandEffector
         brooklyn.config:
-          name: "Login User"
-          description: "Logs in a registered user"
-
-          shell.env:
-            INSTALL_DIR: $brooklyn:attributeWhenReady("install.dir")
+          container.name: "fabric-orderer"
+
+          launch.command: |
+            sudo docker run \
+                --name=${CONTAINER_NAME} \
+                --restart=always \
+                -d \
+                -e ORDERER_GENERAL_LEDGERTYPE=ram \
+                -e ORDERER_GENERAL_BATCHTIMEOUT=10s \
+                -e ORDERER_GENERAL_MAXMESSAGECOUNT=10 \
+                -e ORDERER_GENERAL_MAXWINDOWSIZE=1000 \
+                -e ORDERER_GENERAL_LISTENADDRESS=0.0.0.0 \
+                -e ORDERER_GENERAL_LISTENPORT=7050 \
+                -e ORDERER_RAMLEDGER_HISTORY_SIZE=100 \
+                -e CONFIGTX_ORDERER_ORDERERTYPE=solo \
+                -p 29050:7050 \
+                hyperledger/fabric-orderer:${HYPERLEDGER_VERSION} orderer
+
+    - id: hyperledger-peer-cluster
+      name: "hyperledger-peer-cluster"
+      iconUrl: classpath://io.brooklyn.hyperledger:icon/hyperledger-fabric.png
+      item:
+        type: org.apache.brooklyn.entity.group.DynamicCluster
 
-          command: |
-            timestamp=$(date +%s)
-            rest_endpoint=http://$(cat host-address):7050/registrar
-
-            cp ${INSTALL_DIR}/login-template.json login-${timestamp}.json
-            sed -i "s/ENROLL_ID/${enrollId}/g" login-${timestamp}.json
-            sed -i "s/ENROLL_SECRET/${enrollSecret}/g" login-${timestamp}.json
-
-            curl -H "Content-Type: application/json" -X POST -d @login-${timestamp}.json ${rest_endpoint}
-
-          parameters:
-            enrollId:
-              description: "The username"
-
-            enrollSecret:
-              description: "The password"
-
-      - type: org.apache.brooklyn.core.effector.ssh.SshCommandEffector
         brooklyn.config:
-          name: "Check Login"
-          description: "Checks if a user is logged in"
-
-          command: |
-            rest_endpoint=http://$(cat host-address):7050/registrar/${enrollId}
-            curl ${rest_endpoint}
+          initialSize: $brooklyn:config("hyperledger.num.peers")
+
+          firstMemberSpec:
+            $brooklyn:entitySpec:
+              type: hyperledger-peer-host
+              name: "hyperledger-leader-host"
+              id: my-hyperledger-leader-host
+
+              brooklyn.config:
+                is.leader.node: true
+                latch.launch: $brooklyn:component("my-hyperledger-orderer-node").attributeWhenReady("service.isUp")
+                post.launch.command: |
+                  sudo docker pull hyperledger/fabric-ccenv:${HYPERLEDGER_VERSION}
```

Comment:
`${HYPERLEDGER_VERSION}` is not in the shell.env and so it produces the error `invalid reference format`